### PR TITLE
feat(jmx): add config option for KAFKA_JMX_OPTS

### PR DESCRIPTION
**Pull request by kirecek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | https://github.com/banzaicloud/kafka-operator/issues/351
| License         | Apache 2.0

### What's in this PR?
This PR allows users to configure JMX_PORT and JMX_OPTS environment variables used in https://github.com/kafka-dev/kafka/blob/master/bin/kafka-run-class.sh#L38

### Why?

we would like to collect JMX metrics ourselves since we are using different monitoring stack (DD). See https://github.com/banzaicloud/kafka-operator/issues/351 for more context.

### Checklist

- [x] Implementation tested
- [ ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)

I am totally unsure about the approach and what is the best option here.

NOTE: also set exporter options (jmxImage and pathToBin) as optional since they have hardcoded defaults.

### CODE REVIEW

**Comment by baluchicken** [comment]
Thanks for opening this PR. It is a great improvement to add. I suggest to make it more generic by adding the ability to specify any Env var in the KafkaCluster spec. Moreover, I guess you need to modify the container port used for metrics as well. I think we should make it configurable as well.
Let me know if you have any more questions. I am happy to help.

**Comment by kirecek** [comment]
> I suggest to make it more generic by adding the ability to specify any Env var in the KafkaCluster spec

Actually that was my initial idea, however, then I saw kafkaHeapOpts and JVMOps which are also env variables so I thought this would be conventional approach? 

> need to modify the container port

not required but indeed useful

**Comment by baluchicken** [comment]
Thanks @kirecek LGTM overall, one minor thing, can you please rename the Env field to Envs.

**Comment by kirecek** [comment]
@baluchicken thx, I have updated the field name. I assume json tag should stay the same as the one in container spec, right? :D

**Comment by baluchicken** [comment]
@kirecek I would make the json tag as `envs` as well. Thanks:)

File: pkg/resources/kafka/pod.go [hunk 1]
```diff
@@ -156,7 +174,7 @@ fi
 								},
 							},
 						},
-					},
+					}, r.KafkaCluster.Spec.Envs...),
```
Comment (stoader) [inline pkg/resources/kafka/pod.go:177]: If the user specifies one of the env vars from above (e.g. KAFKA_HEAP_OPTS) in `r.KafkaCluster.Spec.Envs` than the env var list passed to Kafka will contain duplicate env vars.  We should either not allow users to override those env vars via `r.KafkaCluster.Spec.Envs` as for those there is already field in Spec and that should be used or handle duplicates.

I'd opt for the former catching duplicate env vars at validation.

Comment (kirecek) [reply pkg/resources/kafka/pod.go:177]: Thank you.

I had this in my mind actually and that's one of the reasons why I put those settings to monitoringConfig in the first commit https://github.com/banzaicloud/kafka-operator/pull/352/commits/b6accf4a64fc80f0c2530a0134e92d2447ac6ba0 :D

But to be honest, I am not sure how to catch this at validation. Can you please give me some guidance or point me in the right direction before I write something nasty? :D I have checked the kubebuilder validation spec but I don't see there anything useful that I could use for []corev1.EnvVar?

Sorry, I am not experienced much with this :/

Comment (stoader) [reply pkg/resources/kafka/pod.go:177]: for now let's ensure that there are no duplicate env vars and `brokerConfig.GetKafkaHeapOpts()`  has precedence over `Spec.Envs` for `KAFKA_HEAP_OPTS` respectively `brokerConfig.GetKafkaPerfJmvOpts()` for  `KAFKA_JVM_PERFORMANCE_OPTS`.

**Comment by baluchicken** [comment]
Thanks @kirecek

**Pull request merged**